cohttp-eio/src/server.ml

```diff
@@ -65,14 +73,48 @@ let internal_server_error_response =
 let bad_request_response =
   (Http.Response.make ~status:`Bad_request (), Body.Empty)
 
-let write_response ?request writer (response, body) =
+let http_date env =
+  let now = Eio.Time.now env#clock |> Ptime.of_float_s |> Option.get in
+  let (year, mm, dd), ((hh, min, ss), _) = Ptime.to_date_time now in
+  let weekday = Ptime.weekday now in
+  let weekday =
+    match weekday with
+    | `Mon -> "Mon"
+    | `Tue -> "Tue"
+    | `Wed -> "Wed"
+    | `Thu -> "Thu"
+    | `Fri -> "Fri"
+    | `Sat -> "Sat"
+    | `Sun -> "Sun"
+  in
+  let month =
+    match mm with
+    | 1 -> "Jan"
+    | 2 -> "Feb"
+    | 3 -> "Mar"
+    | 4 -> "Apr"
+    | 5 -> "May"
+    | 6 -> "Jun"
+    | 7 -> "Jul"
+    | 8 -> "Aug"
+    | 9 -> "Sep"
+    | 10 -> "Oct"
+    | 11 -> "Nov"
+    | 12 -> "Dec"
+    | _ -> failwith "Invalid HTTP datetime value"
+  in
+  Format.sprintf "%s, %02d %s %04d %02d:%02d:%02d GMT" weekday dd month year hh
```

Comment:
The specification is https://www.rfc-editor.org/rfc/rfc5322#section-3.3

If I am not mistaken `to_date_time` will give the time as UTC/GMT, is that what the env#clock generates or we need to be careful with the timezone?

Reply:
Yes. `Eio.Time.now` uses `Unix.gettimeofday()` so the time is in UTC timezone.